src/modules/referral/services/service.ts

```diff
@@ -172,6 +170,33 @@ export class ReferralService {
     };
   }
 
+  public async getReferralsForDepositsAndUserAddress(depositPrimaryKeys: number[], userAddress: string) {
+    const referrals = await this.depositsMvRepository
+      .createQueryBuilder("d")
+      .where("d.id IN (:...keys)", { keys: depositPrimaryKeys })
+      .getMany();
+
+    return referrals.map((item) => {
+      const appliedRate = new BigNumber(
+        item.depositorAddr === userAddress && item.referralAddress === userAddress
+          ? 1
+          : item.depositorAddr === userAddress
+          ? 0.25
+          : 0.75,
+      )
+        .multipliedBy(item.referralRate)
+        .multipliedBy(item.multiplier);
+      const acxRewards = new BigNumber(item.bridgeFeeUsd)
+        .dividedBy(new BigNumber(item.acxUsdPrice).dividedBy(new BigNumber(10).pow(18)))
+        .multipliedBy(appliedRate);
```

Comment:
I wanted to reuse the existing `getReferralsQuery` in `src/modules/referral/services/queries.ts` but couldn't get the `IN (:...keys)` operator to work with params and raw SQL. So I used this workaround. Happy to change if another solution works better